src/lib.rs

```diff
@@ -224,6 +229,92 @@ pub mod pin {
         }
     }
 
+    /// A pin with software controlled PWM output.
+    pub struct SoftPwmPin<Pin>(libc::c_int, PhantomData<Pin>);
+
+    impl<P: Pin + RequiresRoot> SoftPwmPin<P> {
+        /// Creates a new software PWM controlled pin.
+        ///
+        /// Due to limitations of the chip only one pin is able to do
+        /// hardware-controlled PWM output. The `SoftPwmPin`s on the
+        /// other hand allow for all GPIOs to output PWM signals.
+        ///
+        /// The pulse width of the signal will be 100μs with a value range
+        /// of [0,100] \(where `0` is a constant low and `100` is a
+        /// constant high) resulting in a frequenzy of 100 Hz.
+        ///
+        /// **Important**: In order to use software PWM pins *wiringPi*
+        /// has to be setup in GPIO mode via `setup_gpio()`.
```

Comment:
It may be a good idea to move some of this info to the struct description, to put it at the top of the page, to make it more visible.